src/cloud/functions/leaderboards/around-me/leaderboard/run.csx

```diff
@@ -0,0 +1,160 @@
+    }
+
+    if (string.IsNullOrWhiteSpace(leaderboard) || string.IsNullOrWhiteSpace(playerId))
+        return req.CreateResponse(HttpStatusCode.BadRequest, "Both playerId and leaderboard should be provided in querystring, format is leaderboard/playerId");
+
+    try
+    {
+
+        var scores = client.CreateDocumentQuery<ScoreItem>(
+            UriFactory.CreateDocumentCollectionUri(db, collection), new FeedOptions { EnableCrossPartitionQuery = true });
+
+        // Get score for the current player by playerId        
+        var query =
+            (from s in scores
+             where s.PlayerId == playerId && s.Leaderboard == leaderboard
+             select s);
+
+        var player = query.ToList();
+        if (player.Count < 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"Player with Id = {playerId} in Leaderboard = {leaderboard} does not exist in the database");
+
+        if (player.Count > 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"There are more than one player with Id = {playerId} in Leaderboard = {leaderboard}. This is DB confuguration error - this combination should be unique");
+
+        var currentPlayer = player.First();
+
+        // Calculate global rank for the current player by playerId 
+        int globalRank;
+
+        query = from c in scores
+                where c.Score > currentPlayer.Score && c.Leaderboard == leaderboard
+                select c;
+
+        globalRank = query.Count<ScoreItem>() + 1;
+
+        List<LeaderboardItem> sortedResult = new List<LeaderboardItem>();
+
+        // Get and add to output specific number of players (radiusOfLeaderboard constant) with scores below the current player)
+        query =
+            (from s in scores
+             where s.Score < currentPlayer.Score && s.Leaderboard == leaderboard
+             orderby s.Score descending
+             select s).Take(radiusOfLeaderboard);
+
+        var sortedList = query.ToList<ScoreItem>().OrderBy(a => a.Score);
+
+        foreach (ScoreItem scr in sortedList)
+        {
+            sortedResult.Add(new LeaderboardItem { Rank = 0, Player = scr.Player, Score = scr.Score });
+        }
+        // Add current player to output
+        sortedResult.Add(new LeaderboardItem { Rank = globalRank, Player = currentPlayer.Player, Score = currentPlayer.Score });
+
+        // Get and add to output specific number of players (radiusOfLeaderboard constant) with scores above the current player)
+        query =
+            (from s in scores
+             where s.Score > currentPlayer.Score && s.Leaderboard == leaderboard
+             orderby s.Score ascending
+             select s).Take(radiusOfLeaderboard);
+
+        foreach (ScoreItem scr in query)
+        {
+            sortedResult.Add(new LeaderboardItem { Rank = 0, Player = scr.Player, Score = scr.Score });
+        }
+
+        //Update the ranks in the output with global positions
```

Comment:
The rank calculations currently don't take into account several people with the same score. We need to tweak the algorithm so several players with the same score have the same rank.